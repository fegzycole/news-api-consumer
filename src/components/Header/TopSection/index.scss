@import "../../../styles/mixins";

.TopSection {
  padding: 1rem;
  @include lgScreen {
    padding: 1.5rem 3rem;
  }
}

.Logo {
  display: grid;

  img {
    justify-self: center;
  }
}

.Nav {
  width: 100%;

  @include flex-styles(row, flex-end);
}

.Nav_Item {
  width: 300px;
  margin-top: -20px;
}

.Btn {
  border: none;
  background: $white-color;

  &:hover {
    cursor: pointer;
  }
}

.Actions {
  @include flex-styles("row", space-between);
  width: 100%;
}

.Actions_Item {
  @include text-styles(500, 14px, $primary-color, 20px);
}

.Hide_Sm {
  visibility: hidden;

  @include lgScreen {
    visibility: visible;
  }
}

.Subscribe {
  @include text-styles(500, 14, $white-color);
  background: $secondary-color;
  padding: 4px 16px;
  border-radius: 14px;
  margin-top: -3px;
}
