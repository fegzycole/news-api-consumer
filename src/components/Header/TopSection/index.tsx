import SearchIcon from "../../../assets/Union.svg";
import Logo from "../../../assets/logo.svg";
import Hamburger from "../../Hamburger";

import "./index.scss";

const TopSection = () => {
  return (
    <div className="TopSection">
      <div className="Logo">
        <img src={Logo} alt="Anderson Post Icon" />
      </div>
      <Hamburger />
      <nav className="Nav">
        <ul className="Nav_Item">
          <li className="Actions">
            <button className="Actions_Item Btn Hide_Sm">NewsLetter</button>
            <button className="Actions_Item Btn Hide_Sm">Sign In</button>
            <button className="Subscribe Btn Hide_Sm">Subscribe</button>
            <button className="Search_Icon Btn">
              <img src={SearchIcon} alt="Search Icon" />
            </button>
          </li>
        </ul>
      </nav>
    </div>
  );
};

export default TopSection;
