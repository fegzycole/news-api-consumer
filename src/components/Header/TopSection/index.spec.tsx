import { render } from "@testing-library/react";

import TopSection from "./index";

describe("<TopSection />", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should match the TopSection component snapshot", () => {
    const view = render(<TopSection />);
    expect(view.container).toMatchSnapshot();
  });
});
