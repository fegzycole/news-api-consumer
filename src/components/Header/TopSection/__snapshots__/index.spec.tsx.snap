// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopSection /> should match the TopSection component snapshot 1`] = `
<div>
  <div
    class="TopSection"
  >
    <div
      class="Logo"
    >
      <img
        alt="Anderson Post Icon"
        src="logo.svg"
      />
    </div>
    <span
      class="Hamburger"
    />
    <nav
      class="Nav"
    >
      <ul
        class="Nav_Item"
      >
        <li
          class="Actions"
        >
          <button
            class="Actions_Item Btn Hide_Sm"
          >
            NewsLetter
          </button>
          <button
            class="Actions_Item Btn Hide_Sm"
          >
            Sign In
          </button>
          <button
            class="Subscribe Btn Hide_Sm"
          >
            Subscribe
          </button>
          <button
            class="Search_Icon Btn"
          >
            <img
              alt="Search Icon"
              src="Union.svg"
            />
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
